### PR TITLE
Copter-4.0: add support for ProfiLEDs

### PR DESCRIPTION
**Pull request by IamPete1**
This adds support for ProfiLEDs to Copter 4.0, this also brings the param for the number of notify LED's and fixes the sync loss issue.

This required a couple of other PR's most notably the change of servo function to int16.
#13177
#13535
#13658
#13859
#13783

This is the same thing for plane 4.0 https://github.com/ArduPilot/ardupilot/pull/13996

### CODE REVIEW

File: libraries/AP_Scripting/examples/LED_roll.lua [hunk 1]
```diff
@@ -25,7 +25,8 @@ chan = chan + 1
 gcs:send_text(6, string.format("LEDs: chan=" .. tostring(chan)))
 
 -- initialisation code
-serialLED:set_num_LEDs(chan,  num_leds)
+--serialLED:set_num_neopixel(chan,  num_leds)
```
Comment (rmackay9) [inline libraries/AP_Scripting/examples/LED_roll.lua:28]: I guess we should remove this bit of debug?

**Comment by tridge** [comment]
needs a build all using configure_all.py --build --no-bl

**Comment by rmackay9** [comment]
@IamPete1, can you also check the dshot works?

**Comment by rmackay9** [comment]
After that let's merge

File: libraries/AP_Scripting/examples/LED_roll.lua (hunk 1 above)
Comment (IamPete1) [reply libraries/AP_Scripting/examples/LED_roll.lua:28]: This is just to make it easy for people who have neopixles to change the script, you just have to comment out the ProfiLED one and swap in the neopixel

**Comment by IamPete1** [comment]
done the configure_all.py --build --no-bl , 
Failed builds:
  build: f103-periph

Periph failing is unrelated to these changes
So all of copter passes, do we care that periph fails? Not tested plane, rover ect

**Comment by IamPete1** [comment]
BLHeli works as expected, Suite connects can change settings and run the motor test, Tested on Matekf405

**Comment by rmackay9** [comment]
Merged, thanks!

**Comment by rmackay9** [comment]
This will be included in the Copter-4.0.4-rc1 beta release

**Pull request merged**